File: Code/Java/Palindrome_number.java
/* Check for Palindrome Number 
To check whether the number is palindrome or not ,  
check if the number read backwards is equal to the original number if equal then,
it will be considered to be a palindrome number else it is not a palindrome number.
Algorithm :
Step 1:Get the number to check for palindrome
Step 2: Hold the number in temporary variable (temp)
Step 3: Reverse the number
Step 4: Compare the temporary number with reversed number
Step 5: If both numbers are same, return true
Step 6: Else return false */

package palin;
import java.util.*;
class P
{
	boolean Palindrome (int n)
	{
		//for remainder
		int rem=0;  
		int sum=0;  
		int temp=0;
		//assign n to temp
		temp=n; 
		//loop until n is greater than 0
		while(n>0){ 
			//extract the remainder to get the last digit
			rem= n % 10;   
			sum = (sum*10)+rem; 
			//divide the number by 10
			n = n/10;   
		}
		//original number is equal to reversed number
		if(sum==temp) {
			//true if number is a palindrome
			return true;  
		}
		else 
		{
			//false if number is not a palindrome
			return false; 
		}
	}
}
public class Palindrome_number {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		boolean result; 
		int n1=0;
		//create object of class P
		P pa = new P();  
		System.out.println("-------To check whether a number is palindrome or not-------");
		System.out.println("Enter a number: ");
		n1 = sc.nextInt();
		System.out.println("Input: "+n1);
		result= pa.Palindrome(n1);
		System.out.println("Output: "+result);
	}
}
/*Test case 1: 
-------To check whether a number is palindrome or not------- 
Enter a number: 
124
Input: 124
Output: false

Test case 2:
-------To check whether a number is palindrome or not-------
Enter a number: 
656
Input: 656
Output: true

Time Complexity: O(log10​(n)) where n is the input number
Because we are dividing the number by 10 in every iteration. 
So the time complexity can be said is equal to the number of digits in a number.
Space complexity : O(1) //As Only the number is required to be stored. 
*/

File: Code/Python/palindrome_number.py
# In this problem we have to find whether the given number is a palindrome number or not.
# A palindrome number is a number whose digits remain the same when the number is reversed
# For Ex:
#         INPUT:
#                232
#         OUTPUT:
#                The given number is a palindrome.

#Input

n = int(input("Enter the number\n"))
t = n
rev_no = 0
x = 0

#through the following loop we will find the reverse of the given number

while n != 0:
      c = n%10
      rev_no = (rev_no*10) + (n%10)
      n = n//10
     
#if the given number is equal to its reverse then the given number is a palindrome   

if t == rev_no:
      print("The given number is a palindrome.")
else:
      print("The given number is not a palindrome.")

#         TEST CASES
#      
#     1)INPUT:
#              4325
#       OUTPUT:
#               The given number is not a palindrome.
#
#     2)INPUT:
#              4334
#       OUTPUT:
#               The given number is a palindrome.
    


'''Solution No. 2 Using String ''' 
n = int(input())
n_str = str(n)  #convert it into string in order to do slicing
n_str = int(n_str[::-1]) #reversed the order by slicing

if n_str == n:
      print("The given number is a Palindrome number.")
else:
      print("The given number is not a Palindrome number.")


#TEST CASES : 

'''1. Enter the number : 1252153374328989
The given number is not a Palindrome number.'''

'''2.Enter the number : 23455432
The given number is a Palindrome number.'''


